File: fixtures/php/controller-service-provider/src/PaymentService.php
<?php

namespace ChakraFixture\Service;

use ChakraFixture\Provider\Provider;

final class PaymentService
{
    public function __construct(private Provider $provider)
    {
    }

    public function refund(int $amountCents): void
    {
        $this->provider->refund($amountCents);
    }
}

File: fixtures/go/controller-service-provider/src/payment_service.go
package payments

type Envelope[T any] struct {
	Value T
}

type PaymentService struct {
	Provider PaymentProvider
}

func (service *PaymentService) Refund(amount int) int {
	return providerRefund(amount)
}

func serviceRefund(amount int) int {
	const paymentServiceMarker = 1
	return providerRefund(amount) + paymentServiceMarker - 1
}
